src/Tokenizer/Tokens.php

```diff
@@ -565,25 +589,49 @@ public function getPrevNonWhitespace(int $index, ?string $whitespaces = null): ?
     }
 
     /**
-     * Get index for closest previous token of given kind.
+     * Get index for closest previous token among given kinds.
+     *
      * This method is shorthand for getTokenOfKindSibling method.
      *
-     * @param int $index token index
-     * @param list<array{int}|string|Token> $tokens        possible tokens
-     * @param bool $caseSensitive perform a case sensitive comparison
+     * @param int                                          $index         token index
+     * @param list<array{0: int, 1?: string}|string|Token> $tokens        possible tokens
+     * @param bool                                         $caseSensitive perform a case sensitive comparison
      */
     public function getPrevTokenOfKind(int $index, array $tokens = [], bool $caseSensitive = true): ?int
     {
         return $this->getTokenOfKindSibling($index, -1, $tokens, $caseSensitive);
     }
 
+    /**
+     * Get index for closest previous token of given kind.
+     *
+     * @param int                                        $index         token index
+     * @param array{0: int, 1?: string}|int|string|Token $tokens        possible tokens
+     * @param bool                                       $caseSensitive perform a case sensitive comparison
+     *
+     * @internal
```

Comment:
case 3

not saying to do it right here, right now, but do we aim to deprecate `getPrevTokenOfKind` etc?
overall, I am getting overwhelmed with amount of traverse methods.
if we aim to deprecate, we should prove these new interfaces in battle and use them in our filters, to see if they really fit needs of the fixers.

Reply:
That's exactly my intention: experiment with new methods and maybe later deprecate some of the current ones.

Reply:
let's have https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6680#discussion_r1057999403 aligned first, then